datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaProducerWrapper.java

```diff
@@ -142,7 +143,7 @@ private void populateDefaultProducerConfigs() {
         DEFAULT_MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION_VALUE);
   }
 
-  private Optional<Producer<K, V>> maybeGetKafkaProducer(DatastreamTask task) {
+  protected Optional<Producer<K, V>> maybeGetKafkaProducer(DatastreamTask task) {
```

Comment:
I don't see you overriding this in your Bounded implementation, why make this protected?